src/services/lnd/onchain-service.ts

```diff
@@ -25,19 +25,23 @@ export const OnChainService = (
     try {
       const { current_block_height } = await getHeight({ lnd })
       const after = Math.max(0, current_block_height - scanDepth) // this is necessary for tests, otherwise after may be negative
-      const { transactions } = await getChainTransactions({ lnd, after })
+      const { transactions }: GetChainTransactionsResult = await getChainTransactions({
+        lnd,
+        after,
+      })
 
       return transactions
         .filter((tx) => !tx.is_outgoing || !!tx.transaction)
-        .map((tx) => {
+        .map((tx): SubmittedTransaction => {
           return {
             confirmations: tx.confirmation_count || 0,
             fee: toSats(tx.fee as number),
             id: tx.id as TxId,
             outputAddresses: tx.output_addresses as OnChainAddress[],
             tokens: toSats(tx.tokens),
             rawTx: decoder.decode(tx.transaction as string),
-          } as SubmittedTransaction
+            createdAt: new Date(tx.created_at),
```

Comment:
please use default values for transaction and fee (as confirmation_count) they are optional too https://github.com/alexbosworth/ln-service#getchaintransactions

Reply:
Checking the transaction is in the `filter` statement a few lines above:
```
34       .filter((tx) => !tx.is_outgoing || !!tx.transaction)
```

The `fee` field is never being read so can be ignored - I will remove it to be sure.

Reply:
we use it in onChannelUpdated. We need to keep it if we want to re-use this method

Reply:
I'm just making a commit that derives as much as possible from the raw transaction hex. That we we do not rely on the return of the lnd api.

Reply:
well not sure if i was not clear, the trigger use `onChannelUpdated` to debit fee from ledger `addLndChannelFee`. `onChannelUpdated` use the same code (`getChainTransactions` logic)  to get the fee. If you remove the `fee` then we will need to add it again

Reply:
Yes I understood this and tried to derive it from the txHex. But this isn't possible because the utxo value of the previous tx is not encoded so will just ensure `!!tx.fee` in the filter (like with transaction).